ghcr-signer.py

```diff
@@ -119,25 +120,45 @@ def cli():
 @cli.command()
 @click.argument("image", callback=validate_hash)
 @click.option(
-    "--signatures-dir", default="TO_PUBLISH", help="Base directory to store signatures"
+    "--signatures-dir", default="SIGNATURES", help="Base directory to store signatures"
 )
 @click.option("--key", help="Path to the signing key file")
 @click.option("--sk", is_flag=True, help="Use a hardware security key for signing")
-@click.option("--recursive", is_flag=True)
+@click.option("--recursive", is_flag=True, default=True)
 def prepare(image, signatures_dir, key, sk, recursive):
     """Prepare the signatures for the given IMAGE and saves them to a local folder"""
     ensure_installed()
     with local_registry():
-        prepare_signature(image, signatures_dir, key, sk, recursive, tag=True)
+        prepare_signature(
+            image,
+            signatures_dir,
+            key,
+            sk,
+            recursive,
+            date_folder=None,
+            tag=True,
+        )
 
 
-def prepare_signature(image, signatures_dir, key, sk, recursive, tag=False):
+def prepare_signature(
+    image, signatures_dir, key, sk, recursive, date_folder=None, tag=False
+):
     try:
         signatures_path = Path(signatures_dir)
         signatures_path.mkdir(parents=True, exist_ok=True)
 
+        if not date_folder or not date_folder.exists():
+            today = time.strftime("%Y-%m-%d")
+            date_folder = signatures_path / today
+
+            # Disambiguate the date folder
+            i = 0
+            while date_folder.exists():
+                i = i + 1
+                date_folder = signatures_path / f"{today}-{i}"
```

Comment:
I believe a more robust way, that deals with different timezones (I'm afraid that `time.strftime()` uses naive timezones) and name clashes is this:


```suggestion
                iso_ts = datetime.datetime.now(datetime.UTC).isoformat(timespec="seconds")
                date_folder = signatures_path / iso_ts
```

Reply:
Thanks, I fixed it in 062b67d814fa1303c8b92acffc68343572dc467a